pkg/encoding/sanify.go

```diff
@@ -0,0 +1,49 @@
+package encoding
+
+import (
+	"github.com/pkg/errors"
+	"github.com/surajssd/kapp/pkg/spec"
+)
+
+func fixApp(app *spec.App) error {
+
+	// fix app.Services
+	for i, service := range app.Services {
+		if err := fixService(&service, app); err != nil {
+			return errors.Wrapf(err, "Unable to fix service: %v", service.Name)
+		}
+		app.Services[i] = service
+	}
+
+	// fix app.PersistentVolumes
+	for i, pVolume := range app.PersistentVolumes {
+		if err := fixPersistentVolume(&pVolume, app); err != nil {
+			return errors.Wrapf(err, "Unable to fix persistentVolume: %v", pVolume.Name)
+		}
+		app.PersistentVolumes[i] = pVolume
+	}
+
+	return nil
+}
+
+func fixService(service *spec.ServiceSpecMod, app *spec.App) error {
```

Comment:
so that we do not have to loop over `app.Services` again! Not sure why do we want to increase the complexity when we are getting the services name from `fixApp`